ext/snowcrash/ext/markdown-parser/src/MarkdownNode.h

```diff
@@ -82,13 +82,15 @@ namespace mdp
             const Data& data_ = Data());
 
         /** Copy constructor */
-        MarkdownNode(const MarkdownNode& rhs);
+        MarkdownNode(const MarkdownNode& rhs) = delete;
+        MarkdownNode(MarkdownNode&& rhs) = default;
 
         /** Assignment operator */
-        MarkdownNode& operator=(const MarkdownNode& rhs);
+        MarkdownNode& operator=(const MarkdownNode& rhs) = delete;
```

Comment:
What does this `= delete` and `= default` do?

Reply:
* `MarkdownNode(MarkdownNode&& rhs) = default;` makes the compiler implement the default move constructor for `MarkdownNode`. The default move constructor for structures is what you would expect: move every member of `rhs` to the respective member of the constructed structure.
* `MarkdownNode(const MarkdownNode& rhs)` explicitly denies any declaration of said function; effectively prohibiting copy construction

I figured we don't need to deep-copy our MarkdownNode tree, it made work with ASAN hard because it propagated invalid values from uninitialized memory. Also no reason to deep-copy in our code base.